### PR TITLE
Feature sidebar chat

### PR DESCRIPTION
**Pull request by Travmatth**
Fixes #

## Proposed Changes

  - support sending `chat`, `notification`, `action` type socket events from sidebar
  - support receiving `chat`, `notification`, `action` type socket event in Chat

### CODE REVIEW

File: server/controllers/game.js [hunk 1]
```diff
@@ -77,10 +77,10 @@ const start = (req, res, next) => {
   res.locals.params = [];
 
   const io = req.app.get('socketio');
-  const game = req.locals.game;
+  const game = res.locals.game;
   setInterval(() => {
     game.timerCountDown();
-    io.emit('timer', game.gameState.timer);
+    io.to(req.params.id).emit('update', game.gameState);
   }, 1000);
```
Comment (mpmpx) [inline server/controllers/game.js]: Looks weird. It seems those changes were made in previous commits long ago. This snippet of code in current version of dev looks like: 
https://github.com/hatchways/team-fruit-loops/blob/d246de2bdc5969ca1916b2c12fbe59e06ec23df8/server/controllers/game.js#L79-L88

I am afraid that these changes would overwrite current dev.

Comment (mpmpx) [reply server/controllers/game.js]: It also happens to other files except those changed in your recent commits: d410b50, 8afb3cc, and bc18d04.

**Review by mpmpx — CHANGES_REQUESTED** [review]

File: server/controllers/game.js (hunk 1 above)
Comment (mpmpx) [reply server/controllers/game.js]: Perhaps you merged a previous version of skeleton branch which does not contain latest changes and updates.

Comment (Travmatth) [reply server/controllers/game.js]: I made the branch before the others were created. I'm going to merge dev into my branch, before I merge into dev

File: client/src/pages/Game.js [hunk 2]
```diff
@@ -21,25 +21,25 @@ const styles = theme => ({
...
+//   },
+//   restart: {
+//     url: id => `/game/${id}/restart`,
+//     method: 'PUT',
+//     headers: {
+//       Accept: 'application/json'
+//     },
+//     body: () => ''
+//   }
+// }
```
Comment (mpmpx) [inline client/src/pages/Game.js]: These lines are no longer used, you could remove them before merging.

**Review by mpmpx — APPROVED** [review]
Looks good to me. Good socket handler refactor.

**Review by sina-jamshidi — APPROVED** [review]
Looks good!

**Pull request merged**